File: frontend/src/types/Flashcard.ts
export enum DifficultyLevel {
  Basic = 0,
  Intermediate = 1,
  Advanced = 2,
}

export enum FlashcardCategory {
  AllCategories = -1,
  // Grammatical Categories (starting from 1000 to avoid overlap)
  Verb = 1000,
  Noun = 1001,
  Adjective = 1002,
  Adverb = 1003,
  Pronoun = 1004,
  Connector = 1005,
  // Thematic Categories (starting from 2000)
  Greeting = 2000,
  Phrase = 2001,
  Number = 2002,
  TimeWord = 2003,
  Food = 2004,
  Travel = 2005,
  Family = 2006,
  Work = 2007,
  Nature = 2008,
}

// Add category type enum for navigation
export enum CategoryType {
  DIFFICULTY = "difficulty",
  FLASHCARD_CATEGORY = "flashcard_category",
}

// Helper interface for navigation
export interface CategoryNavigationItem {
  id: string;
  type: CategoryType;
  value: number;
}

export interface FlashcardRequest {
  primaryCategory: FlashcardCategory;
  count: number;
  userId?: string;
  difficulty: DifficultyLevel;
  generateImages: boolean;
  generateAudio: boolean;
}

export interface FlashcardResponse {
  id: string;
  frontText: string;
  backText: string;
  exampleSentence: string;
  exampleSentenceTranslation: string;
  imageUrl: string;
  audioUrl: string;
  notes: string;
  categories: FlashcardCategory[];
  difficulty: DifficultyLevel;
  isVerified: boolean;
}

export interface UpdateFlashcardAdminRequest {
  frontText: string;
  backText: string;
  exampleSentence?: string;
  exampleSentenceTranslation?: string;
  imageUrl?: string;
  audioUrl?: string;
  notes?: string;
  categories: FlashcardCategory[];
  difficulty: DifficultyLevel;
  isVerified: boolean;
}

File: backend/Lithuaningo.API/Models/Flashcard.cs
using System;
using System.Collections.Generic;
using Lithuaningo.API.DTOs.Flashcard;
using Supabase.Postgrest.Attributes;
using Supabase.Postgrest.Models;

namespace Lithuaningo.API.Models
{
    [Table("flashcards")]
    public class Flashcard : BaseModel
    {
        [PrimaryKey("id")]
        [Column("id")]
        public Guid Id { get; set; }

        [Column("front_text")]
        public string FrontText { get; set; } = string.Empty;

        [Column("back_text")]
        public string BackText { get; set; } = string.Empty;

        [Column("example_sentence")]
        public string ExampleSentence { get; set; } = string.Empty;

        [Column("example_sentence_translation")]
        public string ExampleSentenceTranslation { get; set; } = string.Empty;

        [Column("image_url")]
        public string ImageUrl { get; set; } = string.Empty;

        [Column("audio_url")]
        public string AudioUrl { get; set; } = string.Empty;

        [Column("notes")]
        public string Notes { get; set; } = string.Empty;

        [Column("categories")]
        public List<int> Categories { get; set; } = new();

        [Column("difficulty")]
        public DifficultyLevel Difficulty { get; set; } = DifficultyLevel.Basic;

        [Column("is_verified")]
        public bool IsVerified { get; set; } = false;
    }
}
